pkg/k8s/watchers/cilium_network_policy.go

```diff
@@ -82,86 +81,128 @@ func (r *ruleImportMetadataCache) get(cnp *types.SlimCNP) (policyImportMetadata,
 	return policyImportMeta, ok
 }
 
-func (k *K8sWatcher) ciliumNetworkPoliciesInit(cs client.Clientset) {
-	apiGroup := k8sAPIGroupCiliumNetworkPolicyV2
-	_, ciliumV2Controller := informer.NewInformer(
-		utils.ListerWatcherFromTyped[*cilium_v2.CiliumNetworkPolicyList](
-			cs.CiliumV2().CiliumNetworkPolicies("")),
-		&cilium_v2.CiliumNetworkPolicy{},
-		0,
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) {
-				initialRecvTime := time.Now()
-				var valid, equal bool
-				defer func() { k.K8sEventReceived(apiGroup, resources.MetricCNP, resources.MetricCreate, valid, equal) }()
-				if cnp := k8s.ObjToSlimCNP(obj); cnp != nil {
-					valid = true
-					if cnp.RequiresDerivative() {
-						return
-					}
+func (k *K8sWatcher) ciliumNetworkPoliciesInit(ctx context.Context, cs client.Clientset) {
+	var hasSynced atomic.Bool
+	go func() {
```

Comment:
Will this goroutine ever shutdown? Maybe it should also check the context as a signal?

Reply:
I don't think it's needed because the channel returned from `Events` will close as soon as the ctx will be cancelled, and then the goroutine will shutdown .